### PR TITLE
Provisioning: Makes file the default dashboard provisioner type

### LINKED ISSUE #23111: Make file the default dashboard provider type
**What would you like to be added**:

I'd like for the default dashboard provider type to be "file", as this is the only valid value.

**Why is this needed**:

Today, `type` is a required property when defining a dashboard provider, even though there's only [one valid option](https://github.com/grafana/grafana/blob/69c5efcf005cc44e5777aea654800d7a298b94c6/pkg/services/provisioning/dashboards/dashboard.go#L100): `file`:

```
apiVersion: 1

providers:
  - name: Default
    type: file # Should be the default value
    options:
      path: /var/lib/grafana/dashboards
```

Every required property places cognitive load on the person learning something new. We should strive to simplify the number of things they need to know to get started.

### PR DESCRIPTION
**Pull request by bergquist**
closes https://github.com/grafana/grafana/issues/23111

### CODE REVIEW

**Review by marefr — APPROVED** [review]
LGTM, but hard to see what docs changes you've made due to a lot of formatting changes

**Comment by bergquist** [comment]
@marefr yes. vscode went crazy on those files. I'll make sure to split that into multiple commits next time.

**Comment by bergquist** [comment]
I've removed "Required" on the folder setting (cause it's not, it will put dashboards in the root folder if it's missing) and added a note about file being the default for `type`.

File: pkg/services/provisioning/dashboards/config_reader_test.go [hunk 1]
```diff
@@ -4,87 +4,99 @@ import (
...
-	Convey("Dashboards as configuration", t, func() {
+	t.Run("Dashboards as configuration", func(t *testing.T) {
+
 		logger := log.New("test-logger")
 
-		Convey("Can read config file version 1 format", func() {
+		t.Run("default values should be applied", func(t *testing.T) {
+			cfgProvider := configReader{path: appliedDefaults, log: logger}
+			cfg, err := cfgProvider.readConfig()
+			require.Nil(t, err)
```
Comment (aknuds1) [inline pkg/services/provisioning/dashboards/config_reader_test.go:27]:
```suggestion
			require.NoError(t, err)
```

File: pkg/services/provisioning/dashboards/config_reader_test.go [hunk 2]
```diff
@@ -4,87 +4,99 @@ import (
...
+			require.Equal(t, int64(10), cfg[0].UpdateIntervalSeconds)
+		})
+
+		t.Run("Can read config file version 1 format", func(t *testing.T) {
 			_ = os.Setenv("TEST_VAR", "general")
 			cfgProvider := configReader{path: simpleDashboardConfig, log: logger}
 			cfg, err := cfgProvider.readConfig()
 			_ = os.Unsetenv("TEST_VAR")
-			So(err, ShouldBeNil)
+			require.Nil(t, err)
```
Comment (aknuds1) [inline pkg/services/provisioning/dashboards/config_reader_test.go:39]:
```suggestion
			require.NoError(t, err)
```

File: pkg/services/provisioning/dashboards/config_reader_test.go [hunk 3]
```diff
@@ -4,87 +4,99 @@ import (
...
 
 			validateDashboardAsConfig(t, cfg)
 		})
 
-		Convey("Can read config file in version 0 format", func() {
+		t.Run("Can read config file in version 0 format", func(t *testing.T) {
 			cfgProvider := configReader{path: oldVersion, log: logger}
 			cfg, err := cfgProvider.readConfig()
-			So(err, ShouldBeNil)
+			require.Nil(t, err)
```
Comment (aknuds1) [inline pkg/services/provisioning/dashboards/config_reader_test.go:47]:
```suggestion
			require.NoError(t, err)
```

**Review by aknuds1 — APPROVED** [review]
LGTM, just left some suggestions.

**Pull request merged**